src/sql/workbench/parts/objectExplorer/browser/serverTreeRenderer.ts

```diff
@@ -152,19 +153,44 @@ export class ServerTreeRenderer implements IRenderer {
 		let iconLowerCaseName = iconName.toLocaleLowerCase();
 		templateData.icon.classList.add(iconLowerCaseName);
 
+		if (treeNode.iconPath) {
+			renderIcon(templateData.icon, treeNode.iconPath);
+		}
+
 		templateData.label.textContent = treeNode.label;
 		templateData.root.title = treeNode.label;
 	}
 
+	private getIconPath(connection: ConnectionProfile): { light: string, dark: string } {
+		let iconPath = undefined;
+		let serverInfo = this._connectionManagementService.getServerInfo(connection.id);
+		let providerProperties = this._connectionManagementService.getProviderProperties(connection.providerName);
+		if (serverInfo && providerProperties) {
+			try {
+				let icons: { condition: string, path: object }[] = providerProperties['iconPath'];
```

Comment:
Condition should be optional, and ideally this interface would be defined in the connectionProviderExtension.ts and referenced as a clear interface type - with "when" (not condition) being an optional property.

Reply:
fixed.